src/test/scala/io/iohk/ethereum/vm/OpCodeFunSpec.scala

```diff
@@ -251,28 +253,40 @@ class OpCodeFunSpec extends FunSuite with OpCodeTesting with Matchers with Prope
     }
   }
 
-  ignore("BLOCKHASH") {
-    // to be implemented
-  }
+  test(BLOCKHASH) { op =>
+    val stateGen: Gen[PS] = for {
+      extCode <- getByteStringGen(0, 256)
 
-  ignore("COINBASE") {
-    // to be implemented
-  }
+      stateIn <- getProgramStateGen(
+        stackGen = getStackGen(maxWord = DataWord(256)),
+        memGen = getMemoryGen(256),
+        codeGen = getByteStringGen(0, 256)
+      )
 
-  ignore("TIMESTAMP") {
-    // to be implemented
-  }
+      doSave <- Gen.oneOf(false, true, true)
 
-  ignore("NUMBER") {
-    // to be implemented
-  }
+      addr = Address(stateIn.stack.pop._1)
+      hash = kec256(extCode)
+      world = if (doSave) stateIn.world.saveAccount(addr, Account.Empty.copy(codeHash = hash)) else stateIn.world
+    } yield stateIn.withWorld(world)
```

Comment:
To generate various program states  with various world states 😉 
Since it is generative testing I think it is good to have many cases to test and since you approved this PR I assumed that this is correct approach, to check op code when other parts of the system are changing, not only stack and block hashes